src/compiler/parsing/watf/grammar.js

```diff
@@ -159,11 +160,11 @@ export function parse(tokensList: Array<Object>, source: string): Program {
         );
       }
 
-      const limits = t.limits(token.value);
+      const limits = t.limits(parse32I(token.value));
```

Comment:
@xtuc - I've added a fix for the 'limits' bug. Not sure whether I should be importing numeric parsing logic into `grammar.js` - or whether `t.limits` should take string args and perform this parsing logic. Any thoughts?

Reply:
Conceptually t.* shouldn't have too much logic, they're just there to construct the AST.

Reply:
Conceptually t.* shouldn't have too much logic, they're just there to construct the AST.

Reply:
Conceptually t.* shouldn't have too much logic, they're just there to construct the AST.